File: fm-dddrive/src/main/java/io/dddrive/core/doc/model/base/DocRepositoryBase.java
package io.dddrive.core.doc.model.base;

import java.util.Set;

import io.dddrive.core.ddd.model.AggregateRepository;
import io.dddrive.core.ddd.model.base.AggregateRepositoryBase;
import io.dddrive.core.doc.model.Doc;
import io.dddrive.core.doc.model.DocPartTransition;
import io.dddrive.core.doc.model.DocRepository;

public abstract class DocRepositoryBase<D extends Doc>
		extends AggregateRepositoryBase<D>
		implements DocRepository<D> {

	private static final Set<String> NotLoggedProperties = Set.of("docTypeId", "caseDef", "isInWork", "transitionList");

	protected DocRepositoryBase(
			Class<? extends AggregateRepository<D>> repoIntfClass,
			Class<? extends Doc> intfClass,
			Class<? extends Doc> baseClass,
			String aggregateTypeId) {
		super(repoIntfClass, intfClass, baseClass, aggregateTypeId);
	}

	@Override
	public boolean doLogChange(String propertyName) {
		if (NotLoggedProperties.contains(propertyName)) {
			return false;
		}
		return super.doLogChange(propertyName);
	}

	@Override
	public void registerParts() {
		this.addPart(Doc.class, DocPartTransition.class, DocPartTransitionBase.class);
	}

}

File: fm-dddrive/src/main/java/io/dddrive/core/doc/model/base/DocRepositoryBase.kt
package io.dddrive.core.doc.model.base

import io.dddrive.core.ddd.model.AggregateRepository
import io.dddrive.core.ddd.model.base.AggregateRepositoryBase
import io.dddrive.core.doc.model.Doc
import io.dddrive.core.doc.model.DocPartTransition
import io.dddrive.core.doc.model.DocRepository

abstract class DocRepositoryBase<D : Doc>(
	repoIntfClass: Class<out AggregateRepository<D>>,
	intfClass: Class<out Doc>,
	baseClass: Class<out Doc>,
	aggregateTypeId: String,
) : AggregateRepositoryBase<D>(repoIntfClass, intfClass, baseClass, aggregateTypeId),
	DocRepository<D> {

	override fun doLogChange(property: String): Boolean {
		if (NotLoggedProperties.contains(property)) {
			return false
		}
		return super.doLogChange(property)
	}

	override fun registerParts() {
		this.addPart<Doc>(Doc::class.java, DocPartTransition::class.java, DocPartTransitionBase::class.java)
	}

	companion object {
		private val NotLoggedProperties = mutableSetOf<String?>("docTypeId", "caseDef", "isInWork", "transitionList")
	}

}
